feat(ci): front-end linting ci

TheOrionsB opened the pull request:
# Description

Set up Continuous Integration (CI) for the frontend

# Fixes

- fixes #18 

# Changes include

- [x] New feature | Breaking: ESLint is set up with the airbnb and airbnb-typescript configuration
- [x] New feature: Prettier is set up.
- [x] New feature: vitest is set up and its related folders and files are created.

# Breaking changes

The linter test will fail as the codebase is not compliant with it yet.
yarn.lock file was pushed. @vrn-sh/members should run ``yarn install`` to get the correct versions.

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added @vrn-sh/members to the reviewers, or specific members of the team
- [x] I have added the needed labels
- [x] I have tested this code
- [x] I have added / updated tests (unit / functionals / end-to-end / ...)
- [x] I have updated the README and other relevant documents (guides...)

# Additional comments

Will need @bogdzn to set up the github workflow that runs the linter and tests on pull requests

TheOrionsB committed:
feat(web): Linter and prettier added

TheOrionsB committed:
feat(web): Unit testing folder add, vitest add to dependencies, placeholder home test running

TheOrionsB committed:
feat(web): Creating README.md

TheOrionsB committed:
feat(ci): Adding github workflows for unit tests and linting

TheOrionsB committed:
feat(ci): Changing directory to web before running npm

TheOrionsB committed:
feat(ci): Running all commands in one run statement

TheOrionsB committed:
feat(ci): Adding cache-depency-path

TheOrionsB committed:
feat(ci): Pushing yarn.lock

TheOrionsB committed:
feat(ci): Adding empty export to __test__/setup.ts

TheOrionsB commented:
Finally, done. Sorry for all the commits. Couldn't figure out how to test the Github actions without pushing.
Linting fails as it's supposed to considering the codebase is not yet compliant with the linter.

djnnvx commented on .github/workflows/frontend-lint.yml:
```diff
@@ -0,0 +1,33 @@
+
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x, 18.x]
```
Current `node` maintained version is `14.18`. Please see [tool-versions](https://github.com/vrn-sh/erp/blob/18-set-up-front-end-ci/.tool-versions) & [asdf](https://asdf-vm.com/guide/getting-started.html) for more info

djnnvx commented on .github/workflows/frontend-unit-tests.yml:
```diff
@@ -0,0 +1,33 @@
+
+    strategy:
+      matrix:
+        node-version: [14.x, 16.x, 18.x]
```
Same here, version 16.x and 18.x are unneeded

djnnvx commented on web/.eslintrc.cjs:
```diff
@@ -0,0 +1,12 @@
+module.exports = {
```
Please exclude `dist/` folder

djnnvx reviewed (changes requested):
Couple things are missing, but overall good PR. Let's have a call tomorrow night for the missing CI.

TheOrionsB committed:
feat(ci): Bump versions and improving workflows
	- Bumping node to version 16.19.1 LTS
	- Singling out the versions to be tested in github actions
	- Auto fixing fixable lint warnings

TheOrionsB committed:
feat(ci): Moving to yarn

TheOrionsB committed:
feat(ci): Fixing (most) linter warnings/errors

TheOrionsB commented:
safe to be merged:
Team.tsx linter errors while be fixed when #43  will be merged
